bin/runDEXseq.R

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env Rscript
+
+if (!requireNamespace("BiocManager", quietly = TRUE)){
+    install.packages("BiocManager", repos='http://cran.us.r-project.org')
+ }
+if (!require("DRIMSeq")){
+    BiocManager::install("DRIMSeq",update = FALSE, ask= FALSE)
+}
+if (!require("DEXSeq")){
+    BiocManager::install("DEXSeq",update = FALSE, ask= FALSE)
+}
+if (!require("stageR")){
+    BiocManager::install("stageR",update = FALSE, ask= FALSE)
+}
+
+args = commandArgs(trailingOnly=TRUE)
+if (length(args) < 2) {
+  stop("Please input the directory with the transcript level featureCounts results and the sample information file", call.=FALSE)
+} else if (length(args)==2) {
+  # default output file
+  args[3] = "DEXseq_out.txt"
+}
+######### import data ###########
+##Import featureCounts data
+#path <- "~/Downloads/nanorna-bam-master/mod/featureCounts_transcript/"
+path<-args[1]
+count_files<- grep(list.files(path), pattern='tx_', inv=T, value=T)
+#create a dataframe for all samples 
+fullpath<-paste(path,count_files[1],sep='/') 
+count.matrix <- data.frame(read.table(fullpath,sep="\t",header=T)[,c(1,7,8)])
+for(i in 2:length(count_files)){
+  fullpath<-paste(path,count_files[i],sep='/') 
+  samp_df <- read.table(fullpath,sep="\t",header=T)[,c(1,8)]
+  count.matrix<- merge(count.matrix,samp_df,by="Geneid",all=TRUE)
+}
+colnames(count.matrix)[1] <- "feature_id"
+#sample information
+#sampleinfo<-read.table("~/Downloads/nanorna-bam-master/samples_conditions.csv",sep=",",header=T)
+sampleinfo<-read.table(args[2],sep=",",header=T)
+colnames(sampleinfo)[1] <- "sample_id"
+condition_names <- c(levels(sampleinfo$condition))
+lgcolName <- "log2fold"
+for (i in length(condition_names):1){
+  lgcolName <- paste(lgcolName,condition_names[i],sep='_')
+}
+######### Filtering #############
+d <- dmDSdata(counts=count.matrix, samples=sampleinfo)
+# include genes expressed in minimal min_samps_gene_expr samples with min_gene_expr
+# include transcripts expressed in min_samps_feature_expr samples with min_feature_expr;  
+# include transcripts expressed in min_samps_feature_prop samples with min_feature_prop;
+n_samp_gene <- length(sampleinfo$sample_id)/2
+n_samp_feature <- length(sampleinfo$sample_id)/2
+min_count_gene <- 10 
+min_count_feature <- 10
+dFilter <- dmFilter(d,
+                    min_samps_feature_expr = n_samp_feature, 
+                    min_samps_feature_prop = n_samp_feature,
+                    min_samps_gene_expr = n_samp_gene, 
+                    min_feature_expr = min_count_feature,
+                    min_gene_expr = min_count_gene,
+                    min_feature_prop=0.1)
+
+########## DEXSeq #########
+formulaFullModel <- as.formula("~sample + exon + condition:exon")
+#formulaReducedModel <- as.formula("~sample + exon + covariate:exon")
```

Comment:
I think for now, the full model looks fine, but it would be great if we could allow for performing reduced model to adjust for covariates.